tools/ci_build/github/azure-pipelines/android-x86_64-crosscompile-ci-pipeline.yml

```diff
@@ -281,19 +306,12 @@ stages:
       - template: templates/clean-agent-build-directory-step.yml
 
   - job: Test_NNAPI_EP
-    pool:
-      # We need macOS-12 to run the Android emulator for now.
-      # https://github.com/actions/runner-images/issues/7671
-      vmImage: 'macOS-12'
+    pool: onnxruntime-Ubuntu2204-AMD-CPU
```

Comment:
one thought - since the build/test stage is now using the same agent pool, maybe we don't need to split them into separate stages anymore. if they are in a single job, we can also avoid the overhead of uploading/downloading artifacts.

Reply:
Will do it in a seoperate  PR